Update UnrealIRCd config to support 4.x options

drGrove opened the pull request.

drGrove committed:
Update config to support 4.x options

drGrove commented:
Needs hashbang/docker-unrealircd#13. I'm also not sure of a good way to test all of this. Any pointers would be appreciated prior to us merging

RyanSquared commented:
@drGrove with testing, you should be able to set up a VM - or even change the Ansible target to your own system - and run the IRCd locally.

RyanSquared commented:
I can't test it myself right now, I'll try to do so over the weekend.

drGrove commented:
Does ansible have a local runner or a just pretend you're on the server and
put things relative to _here_?

On Thu, Oct 4, 2018, 12:15 Ryan <notifications@github.com> wrote:

> I can't test it myself right now, I'll try to do so over the weekend.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/hashbang/admin-tools/pull/144#issuecomment-427137098>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABX9zufIKz1rPwBkXe54tzL4xc9BOhfiks5uhl5AgaJpZM4XHnZg>
> .
>

RyanSquared commented:
> Does ansible have a local runner

`ansible-playbook -i localhost,` instead of just `ansible-playbook`, but I'm not entirely sure how well it would work on a user system because we hardcode in some file paths which might not exist (like `/home/core`)?

RyanSquared commented:
Ansible also has `{{ ansible_env.HOME }}` which as I understand, means the `$HOME` var for the current user.

benharri commented:
you could also spin up a lxd or docker container on a stretch image and run it against that

drGrove committed:
Move to sync, update conf files to 4.x

Move from copy to sync to allow for recursive copy
Update conf files to 4.x
Move to ansbile_env.HOME

drGrove committed:
Add websocket support

drGrove committed:
Update configs to 4.x

drGrove committed:
Create folder ahead of time

drGrove committed:
Fix weird spaces, add missing semicolons

drGrove committed:
Update numeric to follow 4.x config

drGrove committed:
Update configs

RyanSquared commented on files/anope/services.conf.j2:
```diff
@@ -1,21 +1,21 @@
 {
 	name = "services.host"
-	value = "services.irc.hashbang.sh"
+	value = "services.irc.dannygrove.com"
```
i think we should define the top level domain (and possibly subdomains) as a variable so we don't have this issue per-deployment.

drGrove replied:
Definitely a good idea. This is because I'm testing a bunch of stuff. I'm also now moving towards helm for this stuff, So most of this work is just to make sure i'm using the correct config style.
